internal/app/backend/backend_service.go

```diff
@@ -142,10 +147,67 @@ func (s *backendService) getGRPCClient(config *pb.FunctionConfig_Grpc) (pb.Match
 	return client.(pb.MatchFunctionClient), nil
 }
 
+// nolint: unused
+func (s *backendService) matchesFromHTTPMMF(profile *pb.MatchProfile, client *http.Client, baseURL string, matchChan chan<- *pb.Match, errChan chan<- error) {
+	marshaler := jsonpb.Marshaler{}
+	jsonProfile, err := marshaler.MarshalToString(profile)
+	if err != nil {
+		errChan <- err
+		logger.WithError(err).Error("failed to marshal profile pb to string")
+		return
+	}
+
+	reqBody, err := json.Marshal(map[string]string{"profile": jsonProfile})
+	if err != nil {
+		errChan <- err
+		logger.WithError(err).Error("failed to marshal request body")
+		return
+	}
+
+	req, err := http.NewRequest("post", baseURL+"/v1/matchfunction:run", bytes.NewBuffer(reqBody))
+	if err != nil {
+		errChan <- err
+		logger.WithError(err).Error("failed to create mmf http request")
+		return
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		errChan <- err
```

Comment:
We should generate grpc errors here instead since we have enough context here of what probably happened.
For example in this case the error is retryable since it's probably because TCP connection broke.